### PR TITLE
feat: Add CCTP support (SC-592)

### PR DESCRIPTION
**Pull request by lucas-manuel**

### CODE REVIEW

**Review by hexonaut — CHANGES_REQUESTED** [review]

File: src/MainnetController.sol [hunk 1]
```diff
@@ -248,5 +272,37 @@ contract MainnetController is AccessControl {
...
+    /**********************************************************************************************/
+    /*** Relayer bridging functions                                                             ***/
+    /**********************************************************************************************/
+
+    function transferUSDCToCCTP(uint256 usdcAmount, uint32 destinationDomain)
+        external onlyRole(RELAYER) isActive
+    {
+        bytes32 mintRecipient = mintRecipients[destinationDomain];
+
+        require(mintRecipient != 0, "MainnetController/invalid-mint-recipient");
```
Comment (hexonaut) [inline src/MainnetController.sol:284]: I think a more descriptive error is `MainnetController/domain-not-configured`.

File: test/unit/mainnet-controller/Admin.t.sol [hunk 2]
```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.21;
+
+import "test/unit/UnitTestBase.t.sol";
+
+contract MainnetControllerFreezeTests is UnitTestBase {
```
Comment (hexonaut) [inline test/unit/mainnet-controller/Admin.t.sol:6]: Should be `MainnetControllerAdminTests`.

**Review by hexonaut — CHANGES_REQUESTED** [review]
Some minor changes.

File: src/MainnetController.sol (hunk 1 above)
Comment (lucas-manuel) [reply src/MainnetController.sol:284]: Yeah good call, updated

File: test/unit/mainnet-controller/Admin.t.sol (hunk 2 above)
Comment (lucas-manuel) [reply test/unit/mainnet-controller/Admin.t.sol]: Fixed

**Review by hexonaut — APPROVED** [review]

**Pull request merged**